packages/backend/src/ee/services/ai/agent.ts

```diff
@@ -0,0 +1,124 @@
+import { getGenerateTimeSeriesVizConfig } from './tools/generateTimeSeriesVizConfigTool';
+import { getGetOneLineResult } from './tools/getOneLineResult';
+
+import { getExploreInformationPrompt } from './prompts/exploreInformation';
+import { getSystemPrompt } from './prompts/system';
+import type { AiAgentArgs, AiAgentDependencies } from './types/aiAgent';
+
+export const runAgent = async ({
+    args,
+    dependencies,
+}: {
+    args: AiAgentArgs;
+    dependencies: AiAgentDependencies;
+}): Promise<string> => {
+    const findFields = getFindFields({
+        getExplore: dependencies.getExplore,
+        searchFields: dependencies.searchFields,
+    });
+
+    const generateQueryFilters = getGenerateQueryFilters({
+        getExplore: dependencies.getExplore,
+        promptUuid: args.promptUuid,
+        updatePrompt: dependencies.updatePrompt,
+    });
+
+    const generateBarVizConfig = getGenerateBarVizConfig({
+        updateProgress: dependencies.updateProgress,
+        runMiniMetricQuery: dependencies.runMiniMetricQuery,
+        getPrompt: dependencies.getPrompt,
+        updatePrompt: dependencies.updatePrompt,
+        sendFile: dependencies.sendFile,
+    });
+
+    const generateTimeSeriesVizConfig = getGenerateTimeSeriesVizConfig({
+        updateProgress: dependencies.updateProgress,
+        runMiniMetricQuery: dependencies.runMiniMetricQuery,
+        getPrompt: dependencies.getPrompt,
+        updatePrompt: dependencies.updatePrompt,
+        sendFile: dependencies.sendFile,
+    });
+
+    const generateCsv = getGenerateCsv({
+        updateProgress: dependencies.updateProgress,
+        runMiniMetricQuery: dependencies.runMiniMetricQuery,
+        getPrompt: dependencies.getPrompt,
+        updatePrompt: dependencies.updatePrompt,
+        sendFile: dependencies.sendFile,
+        maxLimit: args.maxLimit,
+    });
+
+    const getOneLineResult = getGetOneLineResult({
+        updateProgress: dependencies.updateProgress,
+        runMiniMetricQuery: dependencies.runMiniMetricQuery,
+        getPrompt: dependencies.getPrompt,
+        updatePrompt: dependencies.updatePrompt,
+    });
+
+    const tools = {
+        findFields,
+        generateQueryFilters,
+        generateBarVizConfig,
+        generateCsv,
+        generateTimeSeriesVizConfig,
+        getOneLineResult,
+    };
+
+    const messageHistory = args.messageHistory.map<CoreMessage>(
+        ({ agent: agentType, message }) => {
+            switch (agentType) {
+                case AiChatAgents.AI:
+                    return { role: 'assistant', content: message };
+                case AiChatAgents.HUMAN:
+                    return { role: 'user', content: message };
+                default:
+                    return assertUnreachable(
+                        agentType,
+                        `unknown agent: ${agentType}`,
+                    );
+            }
+        },
+    );
```

Comment:
you can modify `getChatHistoryFromThreadMessages` method to directly do this and get rid of/replace `AiChatAgents` type. (suggestion, could be out of the scope of this pr)

Reply:
will change in this PR